### ISSUE TITLE
Chance = 0 not working.

### ISSUE THREAD

**Issue by Lykdraft** [kmatheussen/radium#827]
When I remember correctly this should work now...?
(Should read chance value from last voice when 0/zero)

However, Chance = 0 disables the voice, no matter if anything else is 256.

**Comment by kmatheussen** [comment]
It doesn't work for voices, just chance text. I'm not sure how easy
it is to implement for voices.

**Comment by Lykdraft** [comment]
Ah I see. No problem then.

**Comment by kmatheussen** [comment]
The problem is that automations are executed before Note On (it has to be that way), so if you have a note on without chance on line 9, and a note on with chance 50% on line 10, and also a voice chance with chance 0 on line 10, the voice chance will have chance 100%, not 50%. But maybe that's okay? This is probably a little bit confusing for users though. Maybe it is possible to give voice chance automation lower priority than note on though, I have to think about that.

**Comment by kmatheussen** [comment]
> Maybe it is possible to give voice chance automation lower priority than note on though, I have to think about that.

No, that doesn't make sense. Turning voices on/off must be done before actually playing a note. This seems to be a logical problem.

**Comment by Lykdraft** [comment]
The way it is done now is pretty decent once you get your head around it.
(which is actually pretty easy I think.)

**Comment by kmatheussen** [comment]
What about voice chance 0? Should I add the same behavior there? Would it be too confusing, or would it be enough with a note in the help page that says something like this:

"Note that if you set voice chance 0 and chance text 0 on the same line, things will not necessarily happen the way you think. Instead (long complicated explanation)".

**Comment by Lykdraft** [comment]
Hmmm. Hard to say. I would need to see how this works out in "real".
In theory, it sounds too confusing because of the redundancy.

**Comment by Lykdraft** [comment]
And yeah, the explanation is cool.. :)
